src/store/ChartSettingStore.js

```diff
@@ -0,0 +1,77 @@
+import { observable, action, reaction, computed, autorunAsync, when } from 'mobx';
+import MenuStore from './MenuStore';
+
+export default class ChartSettingStore {
+    constructor(mainStore) {
+        this.mainStore = mainStore;
+        this.menu = new MenuStore({getContext: () => this.mainStore.chart.context});
+
+        this.restoreSetting();
+    }
+
+    get context() { return this.mainStore.chart.context; }
+    get stx() { return this.context.stx; }
+
+    onContextReady = () => {};
+
+    @observable view = '';
+    @observable language = '';
+    @observable position = '';
+    @observable theme = '';
+
+    restoreSetting() {
+
+        try {
+            let setting_string = CIQ.localStorage.getItem('setting'),
+                setting = JSON.parse(setting_string !== '' ? setting_string : '{}');
+
+            this.language = setting.language;
+            this.position = setting.position;
+            this.theme = setting.theme;
+
+        }catch (e){
```

Comment:
I'm against empty catch blocks (A quick google will show you why). If you're testing out fine, but we don't want this in production code.